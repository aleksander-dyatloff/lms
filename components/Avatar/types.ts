import { AvatarSizes } from './constants'

export namespace AvatarComponent {
  export interface WrapperProps {
    size: number
  }

  export interface Props {
    src?: string
    alt: string
    size: keyof typeof AvatarSizes
  }
}

export default AvatarComponent
