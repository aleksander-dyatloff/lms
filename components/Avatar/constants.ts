export const AvatarSizes = {
  sm: 30,
  md: 50,
  lg: 70,
}
