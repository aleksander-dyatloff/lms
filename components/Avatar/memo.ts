import { memo } from 'react'

import Avatar from './component'

export default memo(Avatar)
