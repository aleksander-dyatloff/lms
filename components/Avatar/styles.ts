import Element from '@components/Element/styles'
import styled from 'styled-components'

import AvatarComponent from './types'

const Wrapper = styled(Element)<AvatarComponent.WrapperProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: ${({ size }) => size / 1.5}px;
  font-weight: 500;
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;
  border-radius: 50%;
  object-fit: cover;
  object-position: center;
  overflow: hidden;
  background-color: var(--contrastColor);
  color: var(--mainColor);
  user-select: none;
`

export default Wrapper
