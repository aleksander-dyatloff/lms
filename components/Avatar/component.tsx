import Image from 'next/image'
import { FC } from 'react'
import { useTheme } from 'styled-components'

import { AvatarSizes } from './constants'
import Wrapper from './styles'
import AvatarComponent from './types'

const Avatar: FC<AvatarComponent.Props> = ({
  alt,
  src,
  size = 'sm',
  ...restProps
}) => {
  const theme = useTheme()

  return (
    <Wrapper
      theme={theme}
      size={AvatarSizes[size]}
      {...restProps}
    >
      {src ? (
        <Image
          width={AvatarSizes[size]}
          height={AvatarSizes[size]}
          src={src}
          alt={alt}
        />
      ) : (
        <span>{alt[0]}</span>
      )}
    </Wrapper>
  )
}

export default Avatar
