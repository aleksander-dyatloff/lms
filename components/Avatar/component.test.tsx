import 'jest-styled-components'

import { render } from '@testing-library/react'
import { Profiler } from 'react'

import Avatar from './component'

describe('Avatar component', () => {
  test('should match to snapshot', () => {
    const { container } = render(
      <Avatar
        src='http://some'
        alt='Some alt'
      />,
    )

    expect(container).toMatchSnapshot()
  })

  test('first render duration should match to specification', () => {
    render(
      <Profiler
        id='Avatar'
        onRender={(_, __, renderDuration) => {
          expect(renderDuration).toBeLessThanOrEqual(5)
        }}
      >
        <Avatar
          src='http://some'
          alt='Some alt'
        />
      </Profiler>,
    )
  })
})
