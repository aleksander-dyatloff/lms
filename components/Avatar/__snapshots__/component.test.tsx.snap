// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should match to snapshot 1`] = `
.c0 {
  padding: 0;
  margin: 0;
  font: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  list-style: none;
  color: inherit;
  background-color: transparent;
  border: none;
  border-radius: 0;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  cursor: inherit;
  outline: none;
  caret-color: inherit;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 20px;
  font-weight: 500;
  width: 30px;
  height: 30px;
  border-radius: 50%;
  object-fit: cover;
  object-position: center;
  background-color: var(--contrastColor);
  color: var(--mainColor);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <div
    class="c0 c1"
    size="30"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 version=%271.1%27 width=%2730%27 height=%2730%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Some alt"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
</div>
`;
